### PR TITLE
[Refactor][CONV1D] Add dilation support to forward kernel

### LINKED ISSUE #1031: [REFACTOR][CONV1D] add dilation support to forward kernel
### Symptom / Motivation

Part of #853.

`Conv1dFwdOp` and `Conv1dBiasFwdOp` are blocked from manifest promotion because the current forward kernel does not implement the PyTorch-compatible `dilation` parameter. This issue extracts the dilation work from #853 so it can be reviewed independently after the op-layer API normalization and before the larger grouped-convolution refactor.

The current Conv1d kernel computes contiguous kernel-window positions only. For `dilation > 1`, PyTorch spaces kernel taps across the input sequence, which changes both the output length formula and the input index used by each kernel tap.

### Root Cause Analysis

The Conv1d dispatch path does not carry `dilation` into kernel construction or kernel math:

- `Conv1dFwdOp` and `Conv1dBiasFwdOp` do not expose or normalize `dilation`.
- `Conv1dKernel`, `_conv1d_wrapped_kernel`, and `_conv1d_kernel` do not accept `dilation_l`.
- The output length currently follows the non-dilated formula: `(L_in + 2 * padding - kW) // stride + 1`.
- The spatial input index currently uses contiguous kernel taps: `il = ol * stride_l + kw - pad_l`.

For PyTorch-compatible dilation, the output length must become `(L_in + 2 * padding - dilation * (kW - 1) - 1) // stride + 1`, and the spatial input index must become `il = ol * stride_l + kw * dilation_l - pad_l`.

### Related Files

- `tileops/ops/convolution.py`
- `tileops/kernels/convolution.py`
- `tests/ops/test_convolution.py`
- `benchmarks/ops/bench_convolution.py`
- `tileops/ops_manifest.yaml`

## Goal

Add `dilation` support to the Conv1d forward kernel path for both no-bias and bias variants, matching `torch.nn.functional.conv1d` for covered non-grouped cases while keeping the change scoped away from grouped convolution and manifest status promotion.

## Plan

1. Add `dilation` / `dilation_l` to the Conv1d op and kernel construction path: `Conv1dFwdOp`, `Conv1dBiasFwdOp`, `Conv1dKernel`, `_conv1d_wrapped_kernel`, and `_conv1d_kernel`.
2. Normalize scalar and 1-element tuple/list `dilation` values at the op layer, consistent with the existing `stride` and `padding` normalization direction.
3. Update the Conv1d output length formula to include dilation: `(L_in + 2 * padding - dilation * (kW - 1) - 1) // stride + 1`.
4. Update the kernel spatial index from contiguous taps to dilated taps: `il = ol * stride_l + kw * dilation_l - pad_l`.
5. Add focused parity tests against `torch.nn.functional.conv1d` with `dilation=2` for both `Conv1dFwdOp` and `Conv1dBiasFwdOp`.
6. Add at least one benchmark case with `dilation > 1` so the new kernel path has a performance baseline and existing `dilation=1` coverage can be compared for regressions.
7. Run focused validation and keep any manifest status changes out of this issue's implementation PR.

## Constraints

- This issue is linked to #853 and covers only the dilation portion of that breakdown.
- Do not implement grouped convolution in this issue.
- Do not modify `tileops/ops_manifest.yaml` status in this issue.
- Preserve existing scalar-int Conv1d behavior for `dilation=1`.
- Keep benchmark updates limited to accepting the new dilation argument, preserving existing benchmark coverage, and adding focused `dilation > 1` benchmark cases; performance tuning is out of scope.

## Acceptance Criteria

- [ ] `Conv1dFwdOp` and `Conv1dBiasFwdOp` accept `dilation` with default value `1`.
- [ ] 1-element tuple/list `dilation` values are normalized consistently with other Conv1d spatial parameters.
- [ ] `Conv1dKernel`, `_conv1d_wrapped_kernel`, and `_conv1d_kernel` receive and use `dilation_l`.
- [ ] Conv1d output length matches PyTorch for covered `dilation > 1` cases.
- [ ] Conv1d input indexing uses `kw * dilation_l` and matches PyTorch output parity for `dilation=2`.
- [ ] Tests cover both no-bias and bias Conv1d variants with `dilation=2`.
- [ ] Benchmarks include at least one Conv1d case with `dilation > 1`, preferably `dilation=2`, and existing `dilation=1` benchmark coverage is preserved.
- [ ] Existing Conv1d tests continue to pass via `pytest tests/ops/test_convolution.py -vvs`.
- [ ] Manifest validation is run for the Conv1d variants, or any remaining blocker is documented explicitly: `python scripts/validate_manifest.py --check-op Conv1dFwdOp` and `python scripts/validate_manifest.py --check-op Conv1dBiasFwdOp`.

### PR DESCRIPTION
**Pull request by RMLYC**
## Summary

Closes #1031.
Related #853.
Related #1037.

This PR adds PyTorch-compatible `dilation` support to the Conv1d forward kernel path and keeps the change scoped away from grouped convolution and manifest status promotion.

It also aligns the two Conv1d forward variants with their manifest semantics:
- `Conv1dFwdOp` is now the no-bias variant and exposes `forward(input, weight)`.
- `Conv1dBiasFwdOp` is now the bias-required variant and exposes `forward(input, weight, bias)`.

Implementation details:
- Adds `dilation` / `dilation_l` plumbing through `Conv1dFwdOp`, `Conv1dBiasFwdOp`, `Conv1dKernel`, `_conv1d_wrapped_kernel`, and `_conv1d_kernel`.
- Updates Conv1d output length to `(L_in + 2 * padding - dilation * (kW - 1) - 1) // stride + 1`.
- Updates kernel input indexing to `il = ol * stride_l + kw * dilation_l - pad_l`.
- Keeps `groups=1` as the only implemented path while validating the manifest-declared groups constraints and rejecting unsupported grouped Conv1d explicitly.
- Keeps Conv1d spatial params aligned with the manifest by accepting `int | tuple[int]`.

## Test plan

- `python -m py_compile tileops/ops/convolution.py tileops/kernels/convolution.py tests/ops/test_convolution.py benchmarks/ops/bench_convolution.py`
- `python -m ruff check tileops/ops/convolution.py tileops/kernels/convolution.py tests/ops/test_convolution.py benchmarks/ops/bench_convolution.py`
- `python -m pytest tests/ops/test_convolution.py --collect-only -q`
- `python -m pytest tests/ops/test_convolution.py -vvs -k conv1d`
- `python -m pytest benchmarks/ops/bench_convolution.py --collect-only -q -k conv1d_bench`
- `python -m pytest tests --collect-only -q`
- `python scripts/validate_manifest.py --check-op Conv1dFwdOp`
- `python scripts/validate_manifest.py --check-op Conv1dBiasFwdOp`

## Benchmark

Adds a focused Conv1d benchmark case with `dilation=2` while preserving existing `dilation=1` benchmark coverage. The benchmark still uses the existing hand-written benchmark harness; #1037 tracks making Conv1d benchmarks fully manifest-driven so manifest validation no longer emits L4 benchmark warnings.

## Regression

Existing Conv1d `dilation=1` behavior remains covered by the original Conv1d test and benchmark cases. New tests cover `dilation=2`, no-bias `Conv1dFwdOp`, and bias-required `Conv1dBiasFwdOp`.

## Additional context

`python scripts/validate_manifest.py --check-op Conv1dFwdOp` and `python scripts/validate_manifest.py --check-op Conv1dBiasFwdOp` now pass, but still report warnings for missing generated shape/dtype parity methods and the non-manifest-driven benchmark path. The benchmark warning is intentionally tracked separately in #1037.

### CODE REVIEW

File: tileops/ops/convolution.py [hunk 1]
```diff
@@ -127,24 +156,33 @@ def __init__(
...
             padding=padding_tuple,
+            dilation=dilation_tuple,
         )
         self.kernel_size = kernel_size_tuple[0]
         self.stride = stride_tuple[0]
         self.padding = padding_tuple[0]
-        self.has_bias = bias
+        self.dilation = dilation_tuple[0]
+        self.groups = groups
+        self.has_bias = False
```
[the remark this answers is not in the document]
Comment (RMLYC) [reply tileops/ops/convolution.py]: Addressed in commit 64704c5. `Conv1dFwdOp` now stores the internal `_has_bias` value on `self.has_bias` while keeping the public no-bias forward signature unchanged.

File: tileops/ops/convolution.py [hunk 2]
```diff
@@ -158,8 +196,9 @@ def __init__(
             kernel_l=self.kernel_size,
             stride_l=self.stride,
             pad_l=self.padding,
+            dilation_l=self.dilation,
             dtype=dtype,
-            has_bias=bias,
+            has_bias=False,
```
[the remark this answers is not in the document]
Comment (RMLYC) [reply tileops/ops/convolution.py]: Addressed in commit 64704c5. The shared Conv1d kernel initialization now passes `has_bias=_has_bias`, so the no-bias and bias variants use the same setup path.

File: tileops/ops/convolution.py [hunk 3]
```diff
@@ -211,12 +249,72 @@ def __init__(
...
+            l_in=l_in,
+            c_out=c_out,
+            kernel_l=self.kernel_size,
+            stride_l=self.stride,
+            pad_l=self.padding,
+            dilation_l=self.dilation,
+            dtype=dtype,
+            has_bias=True,
+            tune=tune,
         )
```
[the remark this answers is not in the document]
Comment (RMLYC) [reply tileops/ops/convolution.py]: Addressed in commit 64704c5. `Conv1dBiasFwdOp.__init__` now validates the public `bias=True` contract and delegates the common Conv1d setup to `super().__init__(..., _has_bias=True)`, removing the duplicated initialization block.

**Review by Ibuki-wind — CHANGES_REQUESTED** [review]
Trust-boundary verdict: block.
Split this into separate stage-owned PRs: implementation (`tileops/ops`, `tileops/kernels`), test (`tests/`), and benchmark (`benchmarks/`).
After that split, the implementation slice still needs one interface cleanup before it matches the ops design rules.
The dilation indexing and output-length updates in `Conv1dKernel` look internally consistent.

**Review by lcy-seso — APPROVED** [review]

**Review by Ibuki-wind — APPROVED** [review]
